config/defaults.mk

```diff
@@ -367,7 +367,7 @@ GINKGO_DIR = @MFEM_DIR@/../ginkgo/install
 GINKGO_SEARCH_DIR = $(subst @MFEM_DIR@,$(MFEM_DIR),$(GINKGO_DIR))
 GINKGO_BUILD_TYPE=Release
 ifeq ($(MFEM_USE_GINKGO),YES)
-   BASE_FLAGS = -std=c++14
+   BASE_FLAGS = -std=c++17
```

Comment:
This can actually conflict with `BASE_FLAGS = -std=c++14` settings for other packages below (e.g. if both GINKGO and RAJA are enabled)

Right now these checks are spread out, but to address situations like this it is probably best to group them together in the beginning or the end of the file, @nbeams can you do that?

Reply:
Ah, good point! What about something like 4606b27?

Reply:
Looks good, thank you!